lambda/simplif.ml

```diff
@@ -24,26 +24,30 @@ open Debuginfo.Scoped_location
 
 exception Real_reference
 
+let check_function_escape id lfun =
+  (* Check that the identifier is not one of the parameters *)
+  let param_is_id (param, _) = Ident.same id param in
+  assert (not (List.exists param_is_id lfun.params));
+  if Ident.Set.mem id (Lambda.free_variables lfun.body) then
+    raise Real_reference
```

Comment:
If the assertion fails, we can end up reporting a reference as escaping when it is not. This is safe (although technically a regression), but it felt better to simply add the assertion. I also considered a simple test instead of an assertion, but it felt wrong to test for something we don't expect to happen.